### PR TITLE
Example of setting data attributes for image_tag

### PR DESCRIPTION
**Pull request by nishantmodak**
See this question every once in awhile, being asked around.
(Example: http://stackoverflow.com/a/15066860/156775)
- Added an example in documentation
- Added test case

### CODE REVIEW

**Comment by schneems** [comment]
Tests failed

File: actionview/lib/action_view/helpers/asset_tag_helper.rb [hunk 1]
```diff
@@ -205,6 +205,8 @@ def favicon_link_tag(source='favicon.ico', options={})
       #   # => <img alt="Icon" height="32" src="/icons/icon.gif" width="32" />
       #   image_tag("/icons/icon.gif", class: "menu_icon")
       #   # => <img alt="Icon" class="menu_icon" src="/icons/icon.gif" />
+      #   image_tag("/icons/icon.gif", data: {title: 'Rails Application'})
```
Comment (rafaelfranca) [inline actionview/lib/action_view/helpers/asset_tag_helper.rb]: space after the `{` and before the `}`

File: actionview/test/template/asset_tag_helper_test.rb [hunk 2]
```diff
@@ -193,7 +193,8 @@ def url_for(*args)
     %(image_tag("//www.rubyonrails.com/images/rails.png")) => %(<img alt="Rails" src="//www.rubyonrails.com/images/rails.png" />),
     %(image_tag("mouse.png", :alt => nil)) => %(<img src="/images/mouse.png" />),
     %(image_tag("data:image/gif;base64,R0lGODlhAQABAID/AMDAwAAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==", :alt => nil)) => %(<img src="data:image/gif;base64,R0lGODlhAQABAID/AMDAwAAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" />),
-    %(image_tag("")) => %(<img src="" />)
+    %(image_tag("")) => %(<img src="" />),
+    %(image_tag("gold.png", data: {title: 'Rails Application'})) => %(<img data-title="Rails Application" src="/images/gold.png" alt="Gold" />)
```
Comment (rafaelfranca) [inline actionview/test/template/asset_tag_helper_test.rb]: space after the `{` and before the `}`

Comment (nishantmodak) [inline actionview/test/template/asset_tag_helper_test.rb]: Fixed

**Comment by nishantmodak** [comment]
@rafaelfranca Fixed review comments
@schneems The failure is unrelated to this commit, but stems from earlier ones in activesupport

**Pull request merged**